src/stratigraphy/util/predictions.py

```diff
@@ -265,31 +162,232 @@ def from_json(cls, prediction_from_file: dict) -> "OverallFilePredictions":
+
+        Returns:
+            OverallMetricsCatalog: A dictionary that maps a metrics name to the corresponding OverallMetrics object
+        """
+        all_metrics = OverallMetricsCatalog()
+        all_metrics.layer_metrics = get_layer_metrics(self, number_of_truth_values)
+        all_metrics.depth_interval_metrics = get_depth_interval_metrics(self)
+
+        # create predictions by language
+        languages = set(fp.metadata.language for fp in self.file_predictions_list)
+        predictions_by_language = {language: OverallFilePredictions() for language in languages}
+
+        for file_predictions in self.file_predictions_list:
+            language = file_predictions.metadata.language
+            if language in predictions_by_language:
+                predictions_by_language[language].add_file_predictions(file_predictions)
+
+        for language, language_predictions in predictions_by_language.items():
+            language_number_of_truth_values = {
+                prediction.file_name: number_of_truth_values[prediction.file_name]
+                for prediction in language_predictions.file_predictions_list
+            }
+            if language == "de":
+                all_metrics.de_layer_metrics = get_layer_metrics(language_predictions, language_number_of_truth_values)
+                all_metrics.de_depth_interval_metrics = get_depth_interval_metrics(language_predictions)
+            elif language == "fr":
+                all_metrics.fr_layer_metrics = get_layer_metrics(language_predictions, language_number_of_truth_values)
+                all_metrics.fr_depth_interval_metrics = get_depth_interval_metrics(language_predictions)
+
+        logger.info("Macro avg:")
+        logger.info(
+            "F1: %.1f%%, precision: %.1f%%, recall: %.1f%%, depth_interval_accuracy: %.1f%%",
+            all_metrics.layer_metrics.macro_f1() * 100,
+            all_metrics.layer_metrics.macro_precision() * 100,
+            all_metrics.layer_metrics.macro_recall() * 100,
+            all_metrics.depth_interval_metrics.macro_precision() * 100,
+        )
+
+        return all_metrics
+
+    def get_metrics(self, field_key: str, field_name: str) -> OverallMetrics:
+        """Get the metrics for a specific field in the predictions.
+
+        Args:
+            field_key (str): The key to access the specific field in the prediction objects.
+            field_name (str): The name of the field being evaluated.
+
+        Returns:
+            OverallMetrics: The requested OverallMetrics object.
+        """
+        overall_metrics = OverallMetrics()
+
+        for file_prediction in self.file_predictions_list:
+            overall_metrics.metrics[file_prediction.file_name] = getattr(file_prediction, field_key)[field_name]
+
+        return overall_metrics
+
+
+def get_layer_metrics(predictions: OverallFilePredictions, number_of_truth_values: dict) -> OverallMetrics:
```

Comment:
A lot of duplicated methods between this file and `score.py`. Did you mean to remove the methods in score.py, or what is the point of copying this methods into this file?

Reply:
Yes, indeed I was moving code from `score.py` to the prediction file. I missed removing the unused code.